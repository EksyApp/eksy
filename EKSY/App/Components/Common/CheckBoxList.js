import React, {Component} from 'react'
import {CheckBox} from './index'
import {ListView} from 'react-native'
import PropTypes from 'prop-types';
import {FiltersShape, StyleShape} from '../../Utils/PropTypeShapes'

export default class CheckBoxList extends Component {
	
	constructor(props) {
		super(props)
		
		const datasource = new ListView.DataSource({rowHasChanged: (r1,r2) => r1 !== r2})
		
		this.state = {
			dataSource: datasource.cloneWithRows(this.props.data)
		}
		
	}
	
	componentWillReceiveProps(props) {
		if(props.data !== this.props.data) {
			const dataSource = this.state.dataSource.cloneWithRows(props.data || []);
			this.setState({dataSource})
		}
	}
	
	render() {
		return (
				<ListView
						style = {this.props.listStyle}
						dataSource={this.state.dataSource}
						renderRow = {(checkbox) => <CheckBox
								checked={checkbox.checked}
								name={checkbox.name}
								title={this.props.titleKey ? checkbox[this.props.titleKey] : checkbox.stateDescription}
								onPress={(name, checked) => {this.props.onPress(name,checked)}}
						/>}
						enableEmptySections={true}
				/>
		)
	}
	
}

CheckBoxList.propTypes = {
	data: FiltersShape,
	listStyle: StyleShape,
	titleKey: PropTypes.string,
	onPress: PropTypes.func,
}