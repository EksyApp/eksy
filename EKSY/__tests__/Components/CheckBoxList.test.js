import React from 'react';
import CheckBoxList from '../../App/Components/Common/CheckBoxList';
import renderer from 'react-test-renderer';

test('CheckBoxList renders correctly', () => {
    const tree = renderer.create(
        <CheckBoxList data = {[{
            name: "Violence",
            stateDescription: "Show violent content"
        }]}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
