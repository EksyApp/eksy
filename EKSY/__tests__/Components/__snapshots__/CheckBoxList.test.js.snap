// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxList renders correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 1,
    }
  }
  enableEmptySections={true}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={undefined}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(226,227,217)",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          onLongPress={undefined}
          onPress={undefined}
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ï‚–
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                Array [
                  Object {
                    "color": "rgb(89,85,78)",
                  },
                  undefined,
                ],
                undefined,
              ],
            ]
          }
        >
          Show violent content
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
